resources/netregexes.ts

```diff
@@ -31,37 +27,70 @@ const keysThatRequireTranslation = [
   'line',
 ];
 
+type ParseHelperField<
+  T extends TriggerTypes,
+  T2 extends NetFieldsReverse[T],
+  field extends keyof T2
+> = {
+  field: T2[field] extends string ? T2[field] : never;
+  value?: string;
+};
+
+type ParseHelperFields<T extends TriggerTypes> = {
+  [field in keyof NetFieldsReverse[T]]: ParseHelperField<T, NetFieldsReverse[T], field>;
+};
+
+const test: ParseHelperFields<'StartsUsing'> = {
+  0: { field: 'type', value: '20' },
+  1: { field: 'timestamp' },
+  2: { field: 'sourceId' },
+  3: { field: 'source' },
+  4: { field: 'id' },
+  5: { field: 'ability' },
+  6: { field: 'targetId' },
+  7: { field: 'target' },
+  8: { field: 'castTime' },
+  9: { field: 'x' },
+  10: { field: 'y' },
+  11: { field: 'z' },
+  12: { field: 'heading' },
+};
+test;
+
 const parseHelper = <T extends TriggerTypes>(
   params: { timestamp?: string; capture?: boolean } | undefined,
   funcName: string,
-  fields: { [s: string]: Fields },
+  fields: Partial<ParseHelperFields<T>>,
 ): CactbotBaseRegExp<T> => {
   params = params ?? {};
   const validFields: string[] = [];
-  for (const value of Object.values(fields)) {
-    if (typeof value !== 'object')
-      continue;
-    validFields.push(value.field);
+
+  for (const index in fields) {
+    const field = fields[index];
+    if (field)
+      validFields.push(field.field);
   }
+
   Regexes.validateParams(params, funcName, ['capture', ...validFields]);
 
   // Find the last key we care about, so we can shorten the regex if needed.
   const capture = Regexes.trueIfUndefined(params.capture);
   const fieldKeys = Object.keys(fields);
-  let maxKey;
+  let _maxKey: string;
```

Comment:
I think underscore on a variable doesn't mean much to me.  Maybe `_maxKey` => `maxKeyStr` to differentiate what it is? (I guess similar `_key` => `keyStr` below for consistency?)

Reply:
Yeah, I named it with the `_` to match the `_key` variable that was already there. Is there maybe an eslint rule to catch this as well, since we're already using `_` to denote parameters that aren't used?

Reply:
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md looks plausible (but also complicated).  It'd be nice to just enforce camel case as a starting point.